src/qiskit_qec/circuits/css_code.py

```diff
@@ -0,0 +1,465 @@
+            )
+        final_outs = [int(b) for b in output[-1]]
+
+        stabilizer_outs = []
+        for j in range(2):
+            stabilizer_outs.append([])
+            for t in range(self.T):
+                round_outs = []
+                for gs in self._gauges4stabilizers[j]:
+                    out = 0
+                    for g in gs:
+                        out += gauge_outs[j][t][g]
+                    out = out % 2
+                    round_outs.append(out)
+                stabilizer_outs[j].append(round_outs)
+
+        bases = ["x", "z"]
+        j = bases.index(self.basis)
+        final_gauges = []
+        for gauge in self._gauges[j]:
+            out = 0
+            for q in gauge:
+                out += final_outs[-q - 1]
+            out = out % 2
+            final_gauges.append(out)
+        final_stabilizers = []
+        for gs in self._gauges4stabilizers[j]:
+            out = 0
+            for g in gs:
+                out += final_gauges[g]
+            out = out % 2
+            final_stabilizers.append(out)
+        stabilizer_outs[j].append(final_stabilizers)
+
+        stabilizer_changes = []
+        for j in range(2):
+            stabilizer_changes.append([])
+            for t in range(self.T + (bases[j] == self.basis)):
+                stabilizer_changes[j].append([])
+                for e in range(len(stabilizer_outs[j][t])):
+                    if t == 0 and j == bases.index(self.basis):
+                        stabilizer_changes[j][t].append(stabilizer_outs[j][t][e])
+                    else:
+                        stabilizer_changes[j][t].append(
+                            (stabilizer_outs[j][t][e] + stabilizer_outs[j][t - 1][e]) % 2
+                        )
+
+        nodes = []
+        for j in range(2):
+            for t, round_changes in enumerate(stabilizer_changes[j]):
+                for e, change in enumerate(round_changes):
+                    if change == 1:
+                        node = DecodingGraphNode(time=t, qubits=self._stabilizers[j][e], index=e)
+                        node.properties["basis"] = bases[j]
+                        nodes.append(node)
+
+        if self.basis == "x":
+            logicals = self.logical_x
+        else:
+            logicals = self.logical_z
+
+        for index, logical_op in enumerate(logicals):
+            logical_out = 0
+            for q in logical_op:
+                logical_out += final_outs[-q - 1]
+            logical_out = logical_out % 2
+
+            if all_logicals or str(logical_out) != logical:
+                node = DecodingGraphNode(
+                    is_boundary=True,
+                    qubits=logical,
+                    index=index,
+                )
+                node.properties["basis"] = self.basis
+                nodes.append(node)
+
+        return nodes
+
+    def check_nodes(self, nodes, ignore_extra_boundary=False):
```

Comment:
Is this something you plan to implement in the future? If it shouldn't be run, can you raise an exception like "NotImplementedError"?